pkg/build/api/v1beta3/types.go

```diff
@@ -0,0 +1,365 @@
+package v1beta3
+
+import (
+	"time"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api/v1beta3"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+)
+
+// BuildLabel is the key of a Pod label whose value is the Name of a Build which is run.
+const BuildLabel = "build"
+
+// Build encapsulates the inputs needed to produce a new deployable image, as well as
+// the status of the execution and a reference to the Pod which executed the build.
+type Build struct {
+	kapi.TypeMeta   `json:",inline"`
+	kapi.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is all the inputs used to execute the build.
+	Spec BuildSpec `json:"spec,omitempty"`
+
+	// Status is the current status of the build.
+	Status BuildStatus `json:"status,omitempty"`
```

Comment:
if we're doing all this restructuring, don't we ideally want to go ahead and make the internal representation match this?